### PR TITLE
Make traptor process faster

### PR DESCRIPTION
**Pull request by rdempsey**
Overhaul the rule matching to be more performant.

### CODE REVIEW

File: traptor/traptor.py [hunk 1]
```diff
@@ -237,28 +237,56 @@ def _find_rule_matches(self, tweet_dict):
...
+
+            for rule in self.redis_rules:
+                # Get the rule to search for and lowercase it
+                search_str = rule['value']
+                search_str = search_str.lower()
+
+                self.logger.debug("Search string used for the rule match: {}".format(search_str.encode('utf-8')))
+                # Lowercase everything in the dict
+                for k, v in new_dict.iteritems():
+                    if v is not None and isinstance(v, str):
```
Comment (joequery) [inline traptor/traptor.py]: Is v guaranteed to only be `str` and not possibly `unicode`? It's generally safer to use `basestring` to catch both cases.

File: traptor/traptor.py [hunk 2]
```diff
@@ -537,7 +565,7 @@ def main(sentry, stdout, info, debug, delay, id, type, key):
         if sentry:
             client = Client(SENTRY_SECRET)
             client.captureException()
-        logger.errror(e)
+        logger.error(e)
```
Comment (joequery) [inline traptor/traptor.py]: Does this logger helper handle exception messages containing non-ascii characters? Or does that situation never arise?

File: traptor/traptor.py (hunk 1 above)
Comment (jasonrhaas) [inline traptor/traptor.py]: @joequery @rdempsey does `str` not catch the unicode case?

File: traptor/traptor.py (hunk 2 above)
Comment (jasonrhaas) [inline traptor/traptor.py]: @joequery good question.  It probably does not do anything special with non-ascii characters.  I'm not sure if that situation would arise

File: tests/data/follow_tweet.json [hunk 3]
```diff
@@ -1 +1,89 @@
...
+{
```
Comment (jasonrhaas) [inline tests/data/follow_tweet.json]: Why did you take this out of standard json format?  Does everything still read in properly using the python `json.load` method?

File: traptor/traptor.py (hunk 2 above)
Comment (rdempsey) [inline traptor/traptor.py]: @joequery that's a non-issue in this case.

File: tests/data/follow_tweet.json (hunk 3 above)
Comment (rdempsey) [inline tests/data/follow_tweet.json]: It's still compliant JSON, I just made it human readable. It still reads in properly and all tests pass.

File: traptor/traptor.py (hunk 1 above)
Comment (rdempsey) [inline traptor/traptor.py]: @jasonrhaas @joequery unicode is a a different type than "str" so we need to check for it.

Comment (joequery) [inline traptor/traptor.py]: @jasonrhaas 

```python
>>> s = "hi"
>>> u = u"hi"
>>> isinstance(s, str)
True
>>> isinstance(s, unicode)
False
>>> isinstance(u, str)
False
>>> isinstance(u, unicode)
True
>>> isinstance(s, basestring)
True
>>> isinstance(u, basestring)
True
```

**Pull request merged**